internal/operands/data-sources/reconcile.go

```diff
@@ -261,53 +288,94 @@ func (d *dataSources) getDataSourcesAndCrons(request *common.Request) (dataSourc
 	}, nil
 }
 
+type reconcileObjects struct {
+	reconcileDataSource     bool
+	reconcileDataImportCron bool
+}
+
 const dataImportCronLabel = "cdi.kubevirt.io/dataImportCron"
 
-func dataSourceAutoUpdateEnabled(dataSource *cdiv1beta1.DataSource, cronByDataSource map[client.ObjectKey]*ssp.DataImportCronTemplate, request *common.Request) (bool, error) {
-	objectKey := client.ObjectKeyFromObject(dataSource)
-	_, cronExists := cronByDataSource[objectKey]
-	if !cronExists {
-		// If DataImportCron does not exist for this DataSource, auto-update is disabled.
-		return false, nil
+func chooseReconciliationObjects(dataSource *cdiv1beta1.DataSource, cronTemplate *ssp.DataImportCronTemplate, existingCron *cdiv1beta1.DataImportCron, request *common.Request) (reconcileObjects, error) {
+	if existingCron != nil {
+		if cronTemplate != nil && common.CheckOwnerAnnotation(existingCron, request.Instance) {
```

Comment:
What should happen if SSP owns the existing cron but cron template is nil? Will it be removed?

Reply:
Yes. Then the operator assumes that the existing cron was previously created from a cron template by SSP.
Here is the code that removes owned crons that don't have template: https://github.com/kubevirt/ssp-operator/blob/92e6176ec7e65a98d9f8d2325f7ef2bb9c773351/internal/operands/data-sources/reconcile.go#L514-L531